apps/indexer/src/api/controller/historical-balances.controller.ts

```diff
@@ -0,0 +1,103 @@
+import { OpenAPIHono as Hono, createRoute, z } from "@hono/zod-openapi";
+import { Address } from "viem";
+
+import { DaoIdEnum } from "@/lib/enums";
+import { caseInsensitiveEnum } from "../middlewares";
+import {
+  HistoricalBalancesService,
+  HistoricalBalance,
+  HistoricalBalancesRequest,
+} from "../services/historical-balances";
+
+export function historicalBalances(app: Hono) {
+  const service = new HistoricalBalancesService();
+
+  app.openapi(
+    createRoute({
+      method: "get",
+      operationId: "historicalBalances",
+      path: "/historical-balances/{daoId}",
+      summary: "Get historical token balances",
+      description:
+        "Fetch historical token balances for multiple addresses at a specific block number using multicall",
+      tags: ["historical-balances"],
+      request: {
+        params: z.object({
+          daoId: caseInsensitiveEnum(DaoIdEnum),
+        }),
+        query: z.object({
+          addresses: z
+            .string()
+            .min(1, "At least one address is required")
+            .transform((str) =>
+              str.split(",").map((addr) => addr.trim() as Address)
+            ),
+          blockNumber: z.coerce
+            .number()
+            .int()
+            .positive("Block number must be positive"),
+        }),
+      },
+      responses: {
+        200: {
+          description: "Successfully retrieved historical balances",
+          content: {
+            "application/json": {
+              schema: z.object({
+                data: z.array(
```

Comment:
this `data` object isn't really needed, its only adding a layer on the graphql query

Reply:
please remove it